lib/matplotlib/axes/_base.py

```diff
@@ -699,6 +699,10 @@ def __init__(self, fig, rect,
                         rcParams['ytick.major.right']),
             which='major')
 
+    def __init_subclass__(cls, **kwargs):
+        cls.__subclass_uses_cla = 'cla' in cls.__dict__
```

Comment:
I think that will fail if subclass A overrides cla and subclass B of A doesn't?  You perhaps need something more like `'cla' in self.__dict__ or super().__subclass_uses_cla` (well, modulo name mangling and such).  (In fact name mangling may also be a problem for such nested subclasses, I would just stick to a single underscore.)

Reply:
I think the name mangling will save us as at each level it will correctly do the right thing for that class?

Reply:
Actually looking at this again I don't think name mangling will help (not be an extra problem), as the mangled name will always be _Axes__subclass_uses_cla?

Reply:
You're right, the mangling doesn't seem to help anything; I will drop that, and see about parent handling as well.